types/firestore.d.ts

```diff
@@ -663,12 +672,30 @@ declare namespace FirebaseFirestore {
       | {initialOpsPerSecond?: number; maxOpsPerSecond?: number};
   }
 
+  /**
+   * The error thrown when a BulkWriter operation fails. Contains an operation
+   * object that can be used to retry the operation.
+   */
+  export class BulkWriterError extends Error {
+    /** The operation that failed. Can be used to retry the operation. */
+    readonly operation: BulkWriterOperation;
+
+    /** The status code of the error. */
+    readonly code: GrpcStatus;
+
+    /** The error message of the error. */
+    readonly message: string;
```

Comment:
I think we may need to include a retry count/attempt number to let users decide when to give up.